models/task.go

```diff
@@ -181,12 +181,12 @@ func GetMigratingTask(repoID int64) (*Task, error) {
 	return &task, nil
 }
 
-// HasMigratingTask returns if migrating task exist for repo.
-func HasMigratingTask(repoID int64) (bool, error) {
-	return db.GetEngine(db.DefaultContext).Exist(&Task{
-		RepoID: repoID,
-		Type:   structs.TaskTypeMigrateRepo,
-	})
+// HasNonFinishedMigratingTask returns if a non-finished migrating task exist for the repo.
+func HasNonFinishedMigratingTask(repoID int64) (bool, error) {
```

Comment:
Not sure about it. There is also failed/stopped status. Unfinished sounds like that the task didn't end yet.

Reply:
Then, why don't we call it `HasFinishedMigratingTask` and change the logic?

Reply:
The problem with that is that the result of that being `true` would not imply that a task is running currently.

Reply:
We don't want the task to be running anymore, because only when the task has finished we can start fetching the Mirror details from the database. So just inverting the logic will do the job.

Reply:
https://github.com/go-gitea/gitea/pull/19601/commits/ee3dec6124f1b0dc673d7eedb9e22f835cac123f